catalog/core/catalog-core-solr/src/test/java/ddf/catalog/source/solr/SolrFilterDelegateTest.java

```diff
@@ -87,6 +87,28 @@ public void intersectsWithValidJtsWkt() {
     }
 
     @Test
+    public void selfIntersectingPolygon() {
+        String wkt = "POLYGON((0 0, 10 0, 10 20, 5 -5, 0 20, 0 0))";
+        stub(mockResolver.getField("testProperty", AttributeFormat.GEOMETRY, false)).toReturn(
+                "testProperty_geohash_index");
+        SolrQuery query = toTest.contains("testProperty", wkt);
+        assertThat(query.getQuery(),
+                startsWith(
+                        "testProperty_geohash_index:\"Contains(POLYGON ((5 -5, 0 0, 0 20, 10 20, 10 0, 5 -5)))\""));
```

Comment:
Couldn't we also be more direct and assert that it blows up before the repair operation is run and does not blow up afterwards? The modified query is sort of a side effect of the actual goal.